src/dstack/_internal/cli/services/configurators/run.py

```diff
@@ -89,6 +91,42 @@ def apply_configuration(
         self.validate_gpu_vendor_and_image(conf)
         self.validate_cpu_arch_and_image(conf)
 
+        working_dir = conf.working_dir
+        if working_dir is None:
+            # Use the default working dir for the image for tasks and services if `commands`
+            # is not set (emulate pre-0.19.27 JobConfigutor logic), otherwise fall back to
+            # `/workflow`.
+            if isinstance(conf, DevEnvironmentConfiguration) or conf.commands:
+                conf.working_dir = LEGACY_REPO_DIR
+                warn(
+                    "[code]working_dir[/code] is not set."
+                    f" Using legacy working directory [code]{LEGACY_REPO_DIR}[/code]\n\n"
+                    "In future versions the default value will be taken from the image\n"
+                    f"- To keep using legacy working directory, set"
+                    f" [code]working_dir[/code] to [code]{LEGACY_REPO_DIR}[/code]\n"
+                    f"- To prepare for the future default, set [code]working_dir[/code]"
+                    " to the image default working directory (only until the new default"
+                    " takes effect)"
+                )
+        elif not is_absolute_posix_path(working_dir):
+            legacy_working_dir = PurePosixPath(LEGACY_REPO_DIR) / working_dir
+            warn(
+                "[code]working_dir[/code] is relative."
+                f" Using legacy working directory [code]{legacy_working_dir}[/code]\n\n"
+                "Future versions will require absolute path\n"
+                f"To keep using legacy working directory, set"
+                f" [code]working_dir[/code] to [code]{legacy_working_dir}[/code]\n"
+            )
+
+        if conf.repos and conf.repos[0].path is None:
+            warn(
+                "[code]repos[0].path[/code] is not set,"
+                f" using legacy repo path [code]{LEGACY_REPO_DIR}[/code]\n\n"
+                "In future versions [code]path[/code] will be mandatory."
```

Comment:
Will [code]path[/code] be mandatory? I remember we discussed that it can default to `/dstack/repos/{repo_name}` or something like that.

Reply:
> I remember we discussed that it can default to /dstack/repos/{repo_name} or something like that.

Right. I've changed the wording of this warning to:

> repos[0].path is not set, using legacy repo path /workflow
>
> _In a future version the default value will be changed_. To keep using /workflow, explicitly set repos[0].path to /workflow